src/main/preload/externalAPI.ts

```diff
@@ -109,6 +111,12 @@ const desktopAPI: DesktopAPI = {
 
     focusPopout: () => ipcRenderer.send(CALLS_POPOUT_FOCUS),
 
+    openThread: (threadID) => ipcRenderer.send(CALLS_WIDGET_OPEN_THREAD, threadID),
+    onOpenThread: (listener) => createListener(CALLS_WIDGET_OPEN_THREAD, listener),
+
+    openStopRecordingModal: (channelID) => ipcRenderer.send(CALLS_WIDGET_OPEN_STOP_RECORDING_MODAL, channelID),
+    onOpenStopRecordingModal: (listener) => createListener(CALLS_WIDGET_OPEN_STOP_RECORDING_MODAL, listener),
```

Comment:
@devinbinnie @cpoile Since we are here, I was considering making this a bit more generic (e.g. `openCallsModal`) so we could pass a modal ID (string or enum) and at least the channel/call ID which is hopefully sufficient in most cases. That would avoid the need to make desktop changes every time we want to add a modal interaction from the widget.

Thoughts?

Reply:
I'd be okay with that, but I think we'd need to lock down the types of modals that could be opened somewhere, and this should never be used to open a modal that's native to the desktop app. If we're just using it as a relay to the calls plugin from the widget (or vice-versa) then I'm okay with it.

Reply:
Okay. Of course we'd still be handling this event only for Calls windows.

Reply:
I'll be deferring this improvement to the next time we'll need this sort of functionality.